packages/devtools_shared/test/helpers.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+typedef TestDtdConnectionInfo = ({
+  String? uri,
+  String? secret,
+  Process? dtdProcess,
+});
+
+/// Helper method to start DTD for the purpose of testing.
+Future<TestDtdConnectionInfo> startDtd() async {
+  final completer =
+      Completer<({String? uri, String? secret, Process? dtdProcess})>();
+  Process? dtdProcess;
+  try {
+    dtdProcess = await Process.start(
+      Platform.resolvedExecutable,
+      ['tooling-daemon', '--machine'],
+    );
+    dtdProcess.stdout.listen((List<int> data) {
```

Comment:
We should cancel this stream once we've gotten the DTD connection information.

Reply:
added in the finally for the outermost try catch.